be/src/exprs/vectorized/array_functions.cpp

```diff
@@ -107,6 +107,225 @@ ColumnPtr ArrayFunctions::array_append([[maybe_unused]] FunctionContext* context
+        ColumnPtr& result_elements = result_array->elements_column();
+
+        result_offsets.reserve(num_array);
+
+        using ValueType = std::conditional_t<std::is_same_v<ArrayColumn, ElementColumn>, uint8_t,
+                                             typename ElementColumn::ValueType>;
+
+        auto offsets_ptr = offsets.get_data().data();
+        [[maybe_unused]] auto elements_ptr = (const ValueType*)(elements.raw_data());
+        auto targets_ptr = (const ValueType*)(targets.raw_data());
+        auto& first_target = *targets_ptr;
+
+        [[maybe_unused]] auto is_null = [](const NullColumn::Container* null_map, size_t idx) -> bool {
+            return (*null_map)[idx] != 0;
+        };
+
+        uint32_t result_offset = 0;
+        for (size_t i = 0; i < num_array; i++) {
+            size_t offset = offsets_ptr[i];
+            size_t array_size = offsets_ptr[i + 1] - offsets_ptr[i];
+            uint32_t total_found = 0;
+            for (size_t j = 0; j < array_size; j++) {
+                if constexpr (NullableElement && !NullableTarget) {
+                    if (is_null(null_map_elements, offset + j)) {
+                        result_elements->append(elements, offset + j, 1);
+                        continue;
+                    }
+                }
+                if constexpr (!NullableElement && NullableTarget) {
+                    if (is_null(null_map_targets, i)) {
+                        result_elements->append(elements, offset + j, 1);
+                        continue;
+                    }
+                }
+                if constexpr (NullableElement && NullableTarget) {
+                    bool null_element = is_null(null_map_elements, offset + j);
+                    bool null_target = is_null(null_map_targets, i);
+                    if (null_element != null_target) {
+                        result_elements->append(elements, offset + j, 1);
+                        continue;
+                    }
+                    if (null_element) {
+                        total_found++;
+                        continue;
+                    }
+                }
+
+                uint8_t found = 0;
+                if constexpr (std::is_same_v<ArrayColumn, ElementColumn>) {
+                    found = (elements.compare_at(offset + j, i, targets, -1) == 0);
+                } else if constexpr (ConstTarget) {
+                    found = (elements_ptr[offset + j] == first_target);
+                } else {
+                    found = (elements_ptr[offset + j] == targets_ptr[i]);
+                }
+
+                if (found) {
+                    total_found++;
+                } else {
+                    result_elements->append(elements, offset + j, 1);
+                }
+            }
+
+            if (array_size - total_found != 0) {
+                result_offset += array_size - total_found + 1;
+                result_offsets.push_back(result_offset);
+            }
```

Comment:
```suggestion
            if (array_size - total_found != 0) {
                result_offset += array_size - total_found + 1;
            }
            result_offsets.push_back(result_offset);
```
when all elements are removed, empty array should be returned. So the result offset need to be append to results_offsets.

Reply:
ok, i fix it